irohad/migration-tool/migration-tool.cpp

```diff
@@ -0,0 +1,321 @@
+                              RocksDbOptions{FLAGS_rocksdb_path},
+                              log_manager));
+  auto db_context_ =
+      std::make_shared<ametsuchi::RocksDBContext>(std::move(rdb_port));
+  std::shared_ptr<iroha::PendingTransactionStorage> pending_txs_storage_ =
+      nullptr;  // CT-error std::make_shared<PendingTransactionStorageImpl>();
+  std::shared_ptr<shared_model::interface::QueryResponseFactory>
+      query_response_factory_ =
+          std::make_shared<shared_model::proto::ProtoQueryResponseFactory>();
+  std::optional<std::unique_ptr<iroha::ametsuchi::VmCaller>> vm_caller_;
+#if 0 and defined(USE_BURROW)
+  vm_caller_ = std::make_unique<iroha::ametsuchi::BurrowVmCaller>();
+#endif
+  std::optional<std::reference_wrapper<const iroha::ametsuchi::VmCaller>>
+      vm_caller_ref;
+  if (vm_caller_) {
+    vm_caller_ref = *vm_caller_.value();
+  }
+  auto process_block =
+      [](std::shared_ptr<shared_model::interface::Block const> block) {};
+  IROHA_EXPECTED_TRY_GET_VALUE(storage,
+                               initStorage(db_context_,
+                                           pending_txs_storage_,
+                                           query_response_factory_,
+                                           FLAGS_block_store_path,
+                                           vm_caller_ref,
+                                           process_block,
+                                           log_manager->getChild("Storage")));
+  return {storage};
+}
+expected::Result<std::unique_ptr<ametsuchi::BlockStorage>>
+makeFlatFileBlockStorage(std::string const &block_storage_dir) {
+  IROHA_EXPECTED_TRY_GET_VALUE(
+      flat_file,
+      ametsuchi::FlatFile::create(
+          block_storage_dir, log_manager->getChild("FlatFile")->getLogger()));
+  return std::make_unique<ametsuchi::FlatFileBlockStorage>(
+      std::move(flat_file),
+      std::make_shared<shared_model::proto::ProtoBlockJsonConverter>(),
+      log_manager->getChild("FlatFileBlockStorage")->getLogger());
+}
+class FlatBlockQuery : public BlockQueryBase {
+ public:
+  FlatBlockQuery(BlockStorage &block_storage, logger::LoggerPtr log)
+      : BlockQueryBase(block_storage, std::move(log)){};
+  std::optional<int32_t> getTxStatus(
+      const shared_model::crypto::Hash &hash) override {
+    assert(0);
+    return {};
+  }
+};
+expected::Result<void, std::string> restoreWsv() {
+  auto log = log_manager->getChild("FlatBlockQuery")->getLogger();
+  auto wsv_restorer = makeWsvRestorer();
+  IROHA_EXPECTED_TRY_GET_VALUE(storage, makeStorage());
+  IROHA_EXPECTED_TRY_GET_VALUE(
+      flat, makeFlatFileBlockStorage(FLAGS_block_store_path));
+  IROHA_EXPECTED_TRY_GET_VALUE(
+      ledger_state,
+      wsv_restorer->restoreWsv(
+          *storage,
+          false,
+          std::make_shared<FlatBlockQuery>(*flat, log),
+          std::make_shared<InMemoryBlockStorageFactory>()));
+  assert(ledger_state);
```

Comment:
It seems not an assertion case.

Reply:
It is copied from 'application.cpp'. 
We expect it never returns nullptr, because we already have expected::error. The logic has reason.
Resolving.